### PR TITLE
[examples][MLIRSparseTensor] Add example to show how sparse vectoriza…

### PR DESCRIPTION
**Pull request by Avimitin**
…tion rewrite ForOp

### CODE REVIEW

File: examples/MLIRSparseTensor/sparse-tensor-vectorization.mlir [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ],
+  iterator_types = ["parallel"],
+  doc = "x(i) = a(i) * b(i)"
+}
+
+// Example for parallel loop vectorization
+func.func @sparse_mul(%arga: tensor<1024xf32, #SparseVector>,
+                     %argb: tensor<1024xf32>,
+                     %argx: tensor<1024xf32>) -> tensor<1024xf32> {
+  %0 = linalg.generic #trait_mul
```
Comment (linuxlonelyeagle) [inline examples/MLIRSparseTensor/sparse-tensor-vectorization.mlir:19]: `%arga` ,`%argb`, `%argx` should be aligned.

File: examples/MLIRSparseTensor/sparse-tensor-vectorization.mlir [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    affine_map<(i) -> ()>    // x (out)
+  ],
+  iterator_types = ["reduction"],
+  doc = "x += a(i) * b(i)"
+}
+
+// Example for reduction loop vectorization
+func.func @sparse_reduction(%arga: tensor<1024xf32, #SparseVector>,
+                           %argb: tensor<1024xf32>,
+                           %argx: tensor<f32>) -> tensor<f32> {
```
Comment (linuxlonelyeagle) [inline examples/MLIRSparseTensor/sparse-tensor-vectorization.mlir:42]: Same as above

**Comment by zhanghb97** [comment]
@Avimitin Please modify the format according to the comments above.

**Comment by Avimitin** [comment]
Fixed and rebased.

**Comment by zhanghb97** [comment]
Thanks!

**Pull request merged**